src/experimentation-ess/common.js

```diff
@@ -418,6 +418,45 @@ function getObjectByProperty(array, name, value) {
   return array.find((e) => e[name] === value);
 }
 
+/**
+ * returns the date n days ago
+ * @param {number} days
+ * @returns {Date} the date n days ago
+ */
+function getDate(days) {
+  const date = new Date();
+  date.setDate(date.getDate() - days);
+  date.setHours(0, 0, 0, 0);
+  return date;
+}
+
+/**
+ * Updates the experiment start, end dates and status based on the inferred dates
+ * @param {*} experiments
+ */
+function updateExperimentMetadata(experiments) {
+  // get the date 2 days ago to account for timezones
+  const yesterday = getDate(2);
+  for (const experiment of experiments) {
+    experiment.startDate = experiment.startDate || experiment.inferredStartDate;
+    const inferredEndDate = new Date(experiment.inferredEndDate);
+    inferredEndDate.setHours(0, 0, 0, 0);
```

Comment:
Shouldn't we have the end date "inclusive" and set to 24/59/59/999?

Reply:
since there're some edge cases after considering the timezone differences, I've just used yesterday as 2 days ago date, that should cover all the edge cases. `inferredEndDate` and `yesterday` both will have the same hours as we're trying to look at the date rather than hours, so used the hours as `0 0 0 0`.